.github/workflows/integration-tests.yaml

```
@@ -38,12 +38,67 @@ jobs:
         if: ${{ steps.changed-files.outputs.any_changed == 'true' }}
         uses: ./.github/actions/setup-dependencies
 
-      - name: Running Integration tests
+      - name: Build db-migrator image
         if: ${{ steps.changed-files.outputs.any_changed == 'true' }}
-        run: make integration-test VERBOSE=true
+        run: |
+          set -euo pipefail
+          # Get the current commit SHA for tagging
+          COMMIT_SHA="${{ github.sha }}"
+          echo "Building db-migrator image tagged with commit: ${COMMIT_SHA}"
+
+          # Build the db-setup container with commit tag
+          sudo make flightctl-db-setup-container
+
+          # Tag the image with the commit SHA
+          sudo podman tag flightctl-db-setup:latest "flightctl-db-setup:${COMMIT_SHA}"
+
+          echo "MIGRATION_IMAGE=flightctl-db-setup:${COMMIT_SHA}" >> "$GITHUB_ENV"
+
+      - name: Running integration tests in template mode (current)
```

minor: is the current here just an internal note might be good to just remove?

```suggestion
      - name: Running integration tests in template mode
```